lambda-kinesis-lambda/template.yaml

```diff
@@ -0,0 +1,59 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: A demonstration of serverless producer-consumer use case using AWS Lambda and AWS Kinesis Data Stream
+Parameters:
+  KinesisStreamName:
+    Type: String
+    Default: testDataStream
+    Description: Enter a name for the Kinesis Data Stream.
+
+Resources:
+  # Define an AWS Kinesis Data Stream
+  KinesisStream:
+    Type: AWS::Kinesis::Stream
+    Properties:
+      Name: !Ref KinesisStreamName
+      ShardCount: 1
+
+  # Define a Lambda function as an application producer
+  ProducerFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: producer/
+      Runtime: python3.9
+      Handler: producer-function.lambda_handler
+      Environment:
+        Variables:
+          KINESIS_STREAM: !Ref KinesisStreamName
+      Policies:
+      - KinesisCrudPolicy:
+          StreamName: !Ref KinesisStreamName
+
+  # Define a Lambda function as an application consumer
+  ConsumerFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: consumer/
+      Runtime: python3.9
+      Handler: consumer-function.lambda_handler
+      Policies: 
+      - AWSLambdaKinesisExecutionRole      
```

Comment:
Can you give a more specific policy. Like https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-policy-template-list.html#kinesis-stream-read-policy ? where it giving read permissions to just the stream you want and not to all the streams in the account?

Reply:
Changes made as suggested.